tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/LambdaCommon.cs

```diff
@@ -38,7 +41,7 @@ internal static Scope CreatePlaceholderScope(Tracer tracer, NameValueHeadersColl
         return tracer.TracerManager.ScopeManager.Activate(span, false);
     }
 
-    internal static Scope SendStartInvocation(ILambdaExtensionRequest requestBuilder, string data, IDictionary<string, string> context)
+    internal static Scope SendStartInvocation(ILambdaExtensionRequest requestBuilder, string data, ILambdaContext context)
```

Comment:
You should update line 48, as the context written is not the `ILambdaContext` but the `.LambdaContext?.ClientContext?.Custom`